### PR TITLE
Sort Greek and Hebrew characters

### PR DESCRIPTION
**Pull request by oscargus**
## PR summary

Closes #26142 (alternative)

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-docstrings) guidelines

### CODE REVIEW

File: doc/sphinxext/math_symbol_table.py [hunk 1]
```diff
@@ -105,7 +104,9 @@ def render_symbol(sym):
 
     lines = []
     for category, columns, syms in symbols:
-        syms = sorted(list(syms))
+        # Assume that lists are correctly sorted
+        if isinstance(syms, set):
+            syms = sorted(list(syms))
```
Comment (tacaswell) [inline doc/sphinxext/math_symbol_table.py:109]:
```suggestion
            syms = sorted(syms)
```

sets are iterable so `sorted` can take them directly.

Super picky, take it or leave it.

File: doc/sphinxext/math_symbol_table.py [hunk 2]
```diff
@@ -6,17 +6,16 @@
 symbols = [
     ["Lower-case Greek",
      6,
-     r"""\alpha \beta \gamma \chi \delta \epsilon \eta \iota \kappa
-         \lambda \mu \nu \omega \phi \pi \psi \rho \sigma \tau \theta
-         \upsilon \xi \zeta \digamma \varepsilon \varkappa \varphi
-         \varpi \varrho \varsigma \vartheta""".split()],
+     r"""\alpha \beta \gamma \delta \epsilon \zeta \eta \theta \iota \kappa \lambda \mu
+         \nu \xi \pi \rho \sigma \tau \upsilon \phi  \psi  \chi \omega \digamma
```
Comment (tacaswell) [inline doc/sphinxext/math_symbol_table.py:10]:
```suggestion
         \nu \xi \pi \rho \sigma \tau \upsilon \phi \chi \psi \omega \digamma
```

wiki says "..., Φ φ, Χ χ, Ψ ψ, ..."

https://en.wikipedia.org/wiki/Greek_alphabet

File: doc/sphinxext/math_symbol_table.py [hunk 3]
```diff
@@ -6,17 +6,16 @@
...
+         \varepsilon \vartheta \varkappa \varphi \varpi \varrho \varsigma""".split()],
```
Comment (tacaswell) [inline doc/sphinxext/math_symbol_table.py:11]: Do we want to put the `\var` versions with their normal versions or all at the end like tihs?

File: doc/sphinxext/math_symbol_table.py [hunk 4]
```diff
@@ -6,17 +6,16 @@
...
-     8,
-     r"""\Delta \Gamma \Lambda \Omega \Phi \Pi \Psi \Sigma \Theta
-     \Upsilon \Xi \mho \nabla""".split()],
+     6,
+     r"""\Gamma \Delta  \Theta \Lambda \Xi \Pi \Sigma \Upsilon \Phi \Psi \Omega
+      """.split()],
     ["Hebrew",
      6,
-     r"""\aleph \beth \daleth \gimel""".split()],
+     r"""\aleph \beth \gimel \daleth""".split()],
```
Comment (tacaswell) [inline doc/sphinxext/math_symbol_table.py:18]:
```suggestion
     r"""\daleth \gimel \beth \aleph""".split()],
```

I trust @story645 's suggestion that the Hebrew should be ordered right-to-left.

**Comment by tacaswell** [comment]
I think this is a more direct approach than #26142 .  This is a script we control for building docs, if we want them ordered....just put them in order and rely on latin alphabet sorting for the symbols.

**Comment by tacaswell** [comment]
Pragmatically, it is unlikely that the Greek alphabet is going to get any additional letters and the correct sorting on the Greek and Hebrew is not equivalent to the lexigraphical sorting.  Just putting them in order is the least work and requires no maintenance. 

On the other hand, it is possible (and certainly more likely) that we may pick up additional math symbols and the math symbols have no natural sorting, so go with lexical sorting.  I guess we could pre-sort them too, but then if we ever do add a new symbol to the set we would have to worry about keep them in order.

File: doc/sphinxext/math_symbol_table.py (hunk 4 above)
Comment (oscargus) [reply doc/sphinxext/math_symbol_table.py:18]: Ahh, missed that aspect of it.

**Pull request closed without merging**